File: src/main/java/me/rohank05/commands/music/AutoPlayCommand.java
package me.rohank05.commands.music;

import me.rohank05.utilities.command.CommandPermissionCheck;
import me.rohank05.utilities.command.ICommand;
import me.rohank05.utilities.music.PlayerManager;
import me.rohank05.utilities.music.TrackManager;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.entities.MessageEmbed;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;

import java.util.Objects;

public class AutoPlayCommand implements ICommand {
    private final PlayerManager playerManager;
    public AutoPlayCommand(PlayerManager playerManager){
        this.playerManager = playerManager;
    }
    @Override
    public void run(SlashCommandInteractionEvent event) {
        if (!CommandPermissionCheck.checkBasePermission(event)) return;
        if (!CommandPermissionCheck.checkPermission(event, this.playerManager)) return;
        TrackManager trackManager = this.playerManager.getGuildMusicManager(Objects.requireNonNull(event.getGuild())).trackManager;
        String AutoPlayStatus = trackManager.isAutoPlay() ? "Autoplay Disabled" : "Autoplay Enabled";
        trackManager.setAutoPlay(!trackManager.isAutoPlay());
        MessageEmbed embed = new EmbedBuilder().setTitle(AutoPlayStatus).setColor(16760143).build();
        event.getInteraction().getHook().sendMessageEmbeds(embed).queue();
    }

    @Override
    public String getName() {
        return "autoplay";
    }

    @Override
    public String getDescription() {
        return "Enable or Disable autoplaying songs after queue ends";
    }
}

File: src/main/kotlin/com/therohankumar/commands/AutoplayCommand.kt
package com.therohankumar.commands

import com.therohankumar.audio.MusicManagerRegistry
import com.therohankumar.commands.util.CommandUtil
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.build.Commands
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData
import org.springframework.stereotype.Component

@Component
class AutoplayCommand(private val registry: MusicManagerRegistry) : SlashCommand {

    override val name = "autoplay"

    override fun build(): SlashCommandData = Commands.slash(name, "Toggle autoplay (queues recommended tracks when empty)")

    override fun execute(event: SlashCommandInteractionEvent) {
        val manager = CommandUtil.requireSameVoice(event, registry) ?: return
        manager.scheduler.autoplay = !manager.scheduler.autoplay
        val status = if (manager.scheduler.autoplay) "enabled" else "disabled"
        event.replyEmbeds(CommandUtil.embed("Autoplay", "Autoplay is now **$status**.")).queue()
    }
}
